src/rabbit_amqqueue.erl

```diff
@@ -397,6 +397,11 @@ not_found_or_absent_dirty(Name) ->
     end.
 
 with(Name, F, E) ->
+    with(Name, F, E, 1000).
+
+with(_Name, _F, E, 0) ->
+    E(infinite_loop_detected);
```

Comment:
I don't think it's useful - this is still about a bug in some other part of the system. This patch will just allow to detect this condition earlier. Maybe knowing exact time will help to localize this bug.